spec/js/query_params_spec.js

```diff
@@ -1,42 +1,126 @@
-import { describe, it } from 'node:test';
+import { describe, it, beforeEach, afterEach } from 'node:test';
 import assert from 'node:assert';
 import { JSDOM } from 'jsdom';
-import { storeUrlQueryParams, removeUrlQueryParams } from '../../assets/js/query_params.js';
-
-function setupTestEnvironment(url) {
-  const dom = new JSDOM('', { url });
-  global.window = dom.window;
-  global.localStorage = dom.window.localStorage;
-  global.URLSearchParams = dom.window.URLSearchParams;
-  return dom;
-}
+import { storeUrlQueryParams } from '../../assets/js/query_params.js';
 
 describe('URL query parameters', () => {
+  const FIXED_TIMESTAMP = 1724339456350;
+  const CACHE_KEY = 'lastParamsUpdate';
+  let originalDateNow;
+  let jsdom;
+
+  function setTestUrl(newUrl = 'https://test.com/') {
+    const url = new URL(newUrl);
+    global.window.history.replaceState({}, '', url);
+  }
+
+  beforeEach(() => {
+    originalDateNow = Date.now;
+    Date.now = () => FIXED_TIMESTAMP;
```

Comment:
`node:test` has utilities for mocking dates, maybe we could use here instead of replacing the global function? 

https://nodejs.org/api/test.html#dates